components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -11020,6 +11048,16 @@ public static boolean isRecommendationEnabled(String tenantDomain) {
         return false;
     }
 
+    public static void publishEvent(EventPublisherType type, EventPublisherEvent event, String errorMessage) {
+        if (Boolean.parseBoolean(System.getenv("FEATURE_FLAG_REPLACE_EVENT_HUB"))) {
+            try {
+                eventPublishers.get(type).publish(event);
```

Comment:
Let's optimize the code to avoid the map.get for a diff we know at the compilation time :)